src/AutoRest.CSharp/MgmtTest/Generation/ResourceTestWriter.cs

```diff
@@ -168,13 +161,7 @@ protected void WriteMethodTest(Resource resource, MgmtClientOperation clientOper
                         List<string> paramNames = WriteOperationParameters(methodParameters, Enumerable.Empty<Parameter> (), exampleModel);
 
                         _writer.Line();
-                        _writer.Append($"{GetAwait(async && !clientOperation.IsPagingOperation(Context))} {resource.Type.Name.FirstCharToLowerCase()}.{testMethodName}(");
-                        foreach (var paramName in paramNames)
-                        {
-                            _writer.Append($"{paramName},");
-                        }
-                        _writer.RemoveTrailingComma();
-                        _writer.LineRaw(");");
+                        WriteMethodTestInvocation(async, clientOperation.IsPagingOperation(Context) || clientOperation.IsListOperation(Context, out var _), $"{resource.Type.Name.FirstCharToLowerCase()}.{testMethodName}", paramNames);
```

Comment:
I believe the `resource.Type.Name.FirstCharToLowerCase()` here is a local variable name? Could we introduce a local variable here in the generator for this name?